cirq-core/cirq/ops/gate_operation.py

```diff
@@ -334,5 +334,7 @@ def _equal_up_to_global_phase_(
             return False
         return protocols.equal_up_to_global_phase(self.gate, other.gate, atol=atol)
 
+    def equal_gates(self, other):
```

Comment:
Needs docstring. Does this really need to be a function on GateOperation?

Reply:
This should definitely not be a function on `GateOperation` as the functionality below to check equality of underlying gates, should also work for other operation types like `TaggedOperations` etc. which won't happen as of now because the function only exists on GateOperations.

Reply:
Removed.